Intermittent test failure

```
com.yahoo.pulsar.broker.admin.AdminApiTest.testIncrementPartitionsOfTopic(com.yahoo.pulsar.broker.admin.AdminApiTest)

  Run 1: AdminApiTest.testIncrementPartitionsOfTopic Â» ServerSideError Some error occou...

  Run 2: AdminApiTest.testIncrementPartitionsOfTopic:1766 expected [8] but found [12]



2017-06-21 06:43:17,217 - ERROR - [main:TestListener@41] - ------------ Test Failed - com.yahoo.pulsar.broker.admin.AdminApiTest / testIncrementPartitionsOfTopic -- attrs: []

com.yahoo.pulsar.client.admin.PulsarAdminException$ServerSideErrorException: Some error occourred on the server

	at com.yahoo.pulsar.client.admin.internal.BaseResource.getApiException(BaseResource.java:160)

	at com.yahoo.pulsar.client.admin.internal.BaseResource$2.failed(BaseResource.java:109)

	at org.glassfish.jersey.client.JerseyInvocation$7.failed(JerseyInvocation.java:965)

	at org.glassfish.jersey.client.JerseyInvocation$7.completed(JerseyInvocation.java:952)

	at org.glassfish.jersey.client.ClientRuntime.processResponse(ClientRuntime.java:196)

	at org.glassfish.jersey.client.ClientRuntime.access$300(ClientRuntime.java:74)

	at org.glassfish.jersey.client.ClientRuntime$2$1$1.run(ClientRuntime.java:166)

	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)

	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)

	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)

	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)

	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)

	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:340)

	at org.glassfish.jersey.client.ClientRuntime$2$1.response(ClientRuntime.java:164)

	at org.glassfish.jersey.client.internal.HttpUrlConnector$3.run(HttpUrlConnector.java:297)

	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)

	at java.util.concurrent.FutureTask.run(FutureTask.java:266)

	at jersey.repackaged.com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute(MoreExecutors.java:299)

	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)

	at jersey.repackaged.com.google.common.util.concurrent.AbstractListeningExecutorService.submit(AbstractListeningExecutorService.java:50)

	at jersey.repackaged.com.google.common.util.concurrent.AbstractListeningExecutorService.submit(AbstractListeningExecutorService.java:37)

	at org.glassfish.jersey.client.internal.HttpUrlConnector.apply(HttpUrlConnector.java:293)

	at org.glassfish.jersey.client.ClientRuntime$2.run(ClientRuntime.java:180)

	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)

	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)

	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)

	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)

	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)

	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:340)

	at org.glassfish.jersey.client.ClientRuntime$3.run(ClientRuntime.java:208)

	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)

	at java.util.concurrent.FutureTask.run(FutureTask.java:266)

	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)

	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)

	at java.lang.Thread.run(Thread.java:748)

Caused by: javax.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error

	at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:1032)

	at org.glassfish.jersey.client.JerseyInvocation.access$900(JerseyInvocation.java:92)

	... 32 more

2017-06-21 06:43:17,220 - INFO  - [main:TestListener@29] - %%%%%%%%%%%% Before configuration - com.yahoo.pulsar.broker.admin.AdminApiTest / cleanup -- attrs: []

2017-06-21 06:43:17,283 - INFO  - [main:TestListener@29] - %%%%%%%%%%%% Before configuration - com.yahoo.pulsar.broker.admin.AdminApiTest / setup -- attrs: []

2017-06-21 06:43:17,738 - INFO  - [main:TestListener@53] - @@@@@@@@@@@@@ STARTING TEST - com.yahoo.pulsar.broker.admin.AdminApiTest / testIncrementPartitionsOfTopic -- attrs: []

2017-06-21 06:43:18,919 - ERROR - [main:TestListener@41] - ------------ Test Failed - com.yahoo.pulsar.broker.admin.AdminApiTest / testIncrementPartitionsOfTopic -- attrs: []

java.lang.AssertionError: expected [8] but found [12]

	at org.testng.Assert.fail(Assert.java:94)

	at org.testng.Assert.failNotEquals(Assert.java:494)

	at org.testng.Assert.assertEquals(Assert.java:123)

	at org.testng.Assert.assertEquals(Assert.java:370)

	at org.testng.Assert.assertEquals(Assert.java:380)

	at com.yahoo.pulsar.broker.admin.AdminApiTest.testIncrementPartitionsOfTopic(AdminApiTest.java:1766)

	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

	at java.lang.reflect.Method.invoke(Method.java:498)

	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)

	at org.testng.internal.Invoker.invokeMethod(Invoker.java:714)

	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)

	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)

	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)

	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)

	at org.testng.TestRunner.privateRun(TestRunner.java:767)

	at org.testng.TestRunner.run(TestRunner.java:617)

	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)

	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)

	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)

	at org.testng.SuiteRunner.run(SuiteRunner.java:240)

	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)

	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)

	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)

	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)

	at org.testng.TestNG.run(TestNG.java:1057)

	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:132)

	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeMulti(TestNGDirectoryTestSuite.java:193)

	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:94)

	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:147)

	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)

	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)

	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
```

--- a/pulsar-broker/src/test/java/com/yahoo/pulsar/broker/admin/AdminApiTest.java
+++ b/pulsar-broker/src/test/java/com/yahoo/pulsar/broker/admin/AdminApiTest.java
@@ -16,8 +16,8 @@
 package com.yahoo.pulsar.broker.admin;
 
 import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertNotEquals;
 import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotEquals;
 import static org.testng.Assert.assertTrue;
 import static org.testng.Assert.fail;
 
@@ -36,7 +36,6 @@ import java.util.concurrent.TimeUnit;
 import javax.ws.rs.client.InvocationCallback;
 import javax.ws.rs.client.WebTarget;
 
-import org.apache.bookkeeper.test.PortManager;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testng.Assert;
@@ -53,7 +52,6 @@ import com.google.common.collect.Sets;
 import com.google.common.hash.Hashing;
 import com.yahoo.pulsar.broker.PulsarServerException;
 import com.yahoo.pulsar.broker.PulsarService;
-import com.yahoo.pulsar.broker.ServiceConfiguration;
 import com.yahoo.pulsar.broker.auth.MockedPulsarServiceBaseTest;
 import com.yahoo.pulsar.broker.namespace.NamespaceEphemeralData;
 import com.yahoo.pulsar.broker.namespace.NamespaceService;
@@ -67,7 +65,6 @@ import com.yahoo.pulsar.client.admin.PulsarAdminException.PreconditionFailedExce
 import com.yahoo.pulsar.client.admin.internal.LookupImpl;
 import com.yahoo.pulsar.client.admin.internal.PersistentTopicsImpl;
 import com.yahoo.pulsar.client.admin.internal.PropertiesImpl;
-import com.yahoo.pulsar.client.api.Authentication;
 import com.yahoo.pulsar.client.api.ClientConfiguration;
 import com.yahoo.pulsar.client.api.Consumer;
 import com.yahoo.pulsar.client.api.ConsumerConfiguration;
@@ -108,14 +105,13 @@ public class AdminApiTest extends MockedPulsarServiceBaseTest {
 
     private static final Logger LOG = LoggerFactory.getLogger(AdminApiTest.class);
 
+    private MockedPulsarService mockPulsarSetup;
+    
     private PulsarService otherPulsar;
 
     private PulsarAdmin otheradmin;
 
     private NamespaceBundleFactory bundleFactory;
-
-    private final int SECONDARY_BROKER_PORT = PortManager.nextFreePort();
-    private final int SECONDARY_BROKER_WEBSERVICE_PORT = PortManager.nextFreePort();
 
     @BeforeMethod
     @Override
@@ -127,16 +123,10 @@ public class AdminApiTest extends MockedPulsarServiceBaseTest {
 
         // create otherbroker to test redirect on calls that need
         // namespace ownership
-        ServiceConfiguration otherconfig = new ServiceConfiguration();
-        otherconfig.setBrokerServicePort(SECONDARY_BROKER_PORT);
-        otherconfig.setWebServicePort(SECONDARY_BROKER_WEBSERVICE_PORT);
-        otherconfig.setLoadBalancerEnabled(false);
-        otherconfig.setClusterName("test");
-
-        otherPulsar = startBroker(otherconfig);
-
-        otheradmin = new PulsarAdmin(new URL("http://127.0.0.1" + ":" + SECONDARY_BROKER_WEBSERVICE_PORT),
-                (Authentication) null);
+        mockPulsarSetup = new MockedPulsarService();
+        mockPulsarSetup.setup();
+        otherPulsar = mockPulsarSetup.getPulsar();
+        otheradmin = mockPulsarSetup.getAdmin();
 
         // Setup namespaces
         admin.clusters().createCluster("use", new ClusterData("http://127.0.0.1" + ":" + BROKER_WEBSERVICE_PORT));
@@ -149,9 +139,7 @@ public class AdminApiTest extends MockedPulsarServiceBaseTest {
     @Override
     public void cleanup() throws Exception {
         super.internalCleanup();
-
-        otheradmin.close();
-        otherPulsar.close();
+        mockPulsarSetup.cleanup();
     }
 
     @DataProvider(name = "numBundles")
@@ -361,7 +349,11 @@ public class AdminApiTest extends MockedPulsarServiceBaseTest {
     public void brokers() throws Exception {
         List<String> list = admin.brokers().getActiveBrokers("use");
         Assert.assertNotNull(list);
-        Assert.assertEquals(list.size(), 2);
+        Assert.assertEquals(list.size(), 1);
+        
+        List<String> list2 = otheradmin.brokers().getActiveBrokers("test");
+        Assert.assertNotNull(list2);
+        Assert.assertEquals(list2.size(), 1);
 
         Map<String, NamespaceOwnershipStatus> nsMap = admin.brokers().getOwnedNamespaces("use", list.get(0));
         // since sla-monitor ns is not created nsMap.size() == 1 (for HeartBeat Namespace)
@@ -1713,7 +1705,6 @@ public class AdminApiTest extends MockedPulsarServiceBaseTest {
      */
     @Test(dataProvider = "topicName")
     public void testIncrementPartitionsOfTopic(String topicName) throws Exception {
-
         final String subName1 = topicName + "-my-sub 1";
         final String subName2 = topicName + "-my-sub 2";
         final int startPartitions = 4;
@@ -1793,7 +1784,28 @@ public class AdminApiTest extends MockedPulsarServiceBaseTest {
         consumer1.close();
         consumer2.close();
         consumer2.close();
-
     }
 
+    class MockedPulsarService extends MockedPulsarServiceBaseTest {
+        @Override
+        protected void setup() throws Exception {
+            conf.setLoadBalancerEnabled(false);
+            conf.setClusterName("test");
+            super.internalSetup();
+        }
+
+        @Override
+        protected void cleanup() throws Exception {
+            super.internalCleanup();
+        }
+
+        public PulsarService getPulsar() {
+            return pulsar;
+        }
+
+        public PulsarAdmin getAdmin() {
+            return admin;
+        }
+    }
+    
 }

